kythe/cxx/indexer/cxx/KytheGraphObserver.cc

```diff
@@ -870,6 +872,20 @@ void KytheGraphObserver::recordFunctionNode(
   }
 }
 
+void KytheGraphObserver::assignUsr(const NodeId& node, llvm::StringRef usr,
+                                   int byte_size) {
+  usr = usr.take_front(byte_size);
+  auto hash = llvm::SHA1::hash(llvm::arrayRefFromStringRef(usr));
```

Comment:
No sure it is intended.
USR length is not fixed, and could be very long for functions with long type arguments.

`SHA1` value is always 20 bytes, truncating the USR doesn't help to reduce the length of the tickets, I think a more reasonable way is to truncate the hash value, which is also what clangd index currently does.